gym/lib/gym/runner.rb

```diff
@@ -22,7 +22,7 @@ def run
 
       if Gym.project.ios? || Gym.project.tvos?
         fix_generic_archive # See https://github.com/fastlane/fastlane/pull/4325
-        return BuildCommandGenerator.archive_path if Gym.config[:skip_package_ipa]
+        return BuildCommandGenerator.archive_path if Gym.config[:skip_package_ipa] || (!Gym.config[:skip_build_archive] && Gym.config[:skip_archive])
```

Comment:
I'm still a little bit unsure about this one 🤔 I think I'm still in favor of...

```rb
return nil if Gym.config[:skip_archive]

if Gym.project.ios? || Gym.project.tvos?
```

There isn't an archive path if we are skipping archive so `nil` should be returned. This should be the case fo both iOS and macOS.

I also think someone could specific `skip_build_archive: true` and `skip_archive: true` (for whatever reason) and use `gym`. This would call `verify_archive` which would be okay and then it return nil because it shouldn't go into any building of anything.

This is getting complicated 😇  Thoughts?

Reply:
Right it is simpler that way. Thanks!

Reply:
Not a problem! Thanks for making changes 😊 I gotta drive to a holiday party now but I will review this tomorrow and most likely get it merged as well 💪